### PR TITLE
CXX-723: make find actually use sort

### PR DESCRIPTION
**Pull request by Machyne**

### CODE REVIEW

File: src/mongocxx/collection.cpp [hunk 1]
```diff
@@ -98,11 +98,15 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
     scoped_bson_t filter_bson;
     scoped_bson_t projection(options.projection());
 
-    if (options.modifiers()) {
-        filter_builder << "$query" << bsoncxx::types::b_document{filter}
-                       << builder::stream::concatenate{
-                              options.modifiers().value_or(document::view{})};
-
+    if (options.sort() || options.modifiers()) {
+        filter_builder << "$query" << bsoncxx::types::b_document{filter};
```
Comment (samantharitter) [inline src/mongocxx/collection.cpp]: Is there a reason we don't want to use $query when we don't have any modifiers?  I'm wondering if we can simplify the logic here.

File: src/mongocxx/collection.cpp [hunk 2]
```diff
@@ -98,11 +98,15 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
...
+        if (options.modifiers()) {
+            filter_builder << builder::stream::concatenate{
+                options.modifiers().value_or(document::view{})};
```
Comment (samantharitter) [inline src/mongocxx/collection.cpp]: We don't need both `if (options.modifiers()) ...` and `options.modifiers().value_or(...)`, I'd say just use the if statement and concatenate `*options.modifiers()`

File: src/mongocxx/collection.cpp [hunk 3]
```diff
@@ -98,11 +98,15 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
...
+        }
+        if (options.sort()) {
+            filter_builder << "$orderby" << bsoncxx::types::b_document{*options.sort()};
```
Comment (samantharitter) [inline src/mongocxx/collection.cpp]: this is a nice alternative to
```
builder << "$orderby" << builder::stream::open_document << builder::stream::concatenate{*options.sort()} << builder::stream::close_document;
```
we should try to do this elsewhere.

Comment (Machyne) [inline src/mongocxx/collection.cpp]: Oh yeah, this is what I wanted in my comment on your most recent PR! I'll add this comment there.

File: src/mongocxx/collection.cpp (hunk 1 above)
Comment (Machyne) [inline src/mongocxx/collection.cpp]: No reason, AFAIK. Done!

**Comment by samantharitter** [comment]
LGTM

File: src/mongocxx/test/collection_mocked.cpp [hunk 4]
```diff
@@ -244,7 +244,8 @@ TEST_CASE("Collection", "[collection]") {
             REQUIRE(skip == skip);
             REQUIRE(limit == limit);
             REQUIRE(batch_size == batch_size);
-            REQUIRE(bson_get_data(query) == doc.data());
+            bsoncxx::document::view query_view{bson_get_data(query), query->len};
```
Comment (acmorrow) [inline src/mongocxx/test/collection_mocked.cpp]: How is this change related to the others?

**Comment by acmorrow** [comment]
LGTM with a question

File: src/mongocxx/test/collection_mocked.cpp (hunk 4 above)
Comment (Machyne) [inline src/mongocxx/test/collection_mocked.cpp]: For simplicity, we now send all finds with the `{$query: {}}` form. Previously we would simply send the match document when we didn't have modifiers to the find. This test assumed that we would never send finds with the $query syntax, and would have broken had we ever tested sorts (which this PR implements) or modifiers.

Since we _always_ use the $query form now, this new assumption is safe (and works when testing sorts). I will add a test to the mock as well to ensure this stays functioning.

**Comment by Machyne** [comment]
@acmorrow - I added a test for sorting in the mock. Hopefully that and my explanation answer your question. Good to merge?

**Comment by acmorrow** [comment]
LGTM
